lesson28/src/js/register.js

```diff
@@ -111,5 +110,17 @@ const saveUserData = () => {
         password : password.value,
     }
 
-    window.localStorage.setItem("userData", JSON.stringify(userData));
+    if( isRegisteredEmail(email) ){
+        validation.showErrorMessage(email,"登録済みのメールアドレスです");
+    }else{
+        window.localStorage.setItem("userData", JSON.stringify(userData));
+        window.location.href = "register-done.html";
+    }
+}
```

Comment:
[nits]
Since only email.value is used in the isRegisteredEmail function, I thought if I could pass the value here.

```suggestion
    if( isRegisteredEmail(userData.email) ){
        validation.showErrorMessage(email,"登録済みのメールアドレスです");
    }else{
        window.localStorage.setItem("userData", JSON.stringify(userData));
        window.location.href = "register-done.html";
    }
}
```

Reply:
Thank you! You are right, it is smarter to just give VALUE.
I thought it was hard to see at first glance that the userData.email is `email.value`, so how about passing `email.value`?
```JavaScript
    if( isRegisteredEmail(email.value) ){
        validation.showErrorMessage(email,"登録済みのメールアドレスです");
    }else{
        window.localStorage.setItem("userData", JSON.stringify(userData));
        window.location.href = "register-done.html";
    }
}
```
and  Align the caller and arguments.

```JavaScript
const isRegisteredEmail = (emailValue) => {
    const userData = JSON.parse(localStorage.getItem("userData"));
    const registeredEmail = userData?.email;

    return registeredEmail === emailValue;
}
```

Reply:
Thanks for the correction!
I thought it would be clearer to pass the email.value! Thanks for the study!